go/database/mpt/proof.go

```diff
@@ -297,3 +298,121 @@ func createPathFromKeyPrefix(key common.Key, nibbles uint8) Path {
+// The method may return an error if the proof is invalid.
+func (c *mptWitnessProofState) GetBalance(address common.Address) (common.Balance, bool, error) {
+	return c.proof.GetBalance(c.root, address)
+}
+
+// GetNonce extracts a nonce from the witness proof for the input root hash and the address.
+// If the witness proof contains the account for the input address, it returns its nonce.
+// If the proof does not contain the account, it returns false.
+// The method may return an error if the proof is invalid.
+func (c *mptWitnessProofState) GetNonce(address common.Address) (common.Nonce, bool, error) {
+	return c.proof.GetNonce(c.root, address)
+}
+
+// GetCodeHash extracts a code hash from the witness proof for the input root hash and the address.
+// If the witness proof contains the account for the input address, it returns its code hash.
+// If the proof does not contain the account, it returns false.
+// The method may return an error if the proof is invalid.
+func (c *mptWitnessProofState) GetCodeHash(address common.Address) (common.Hash, bool, error) {
+	return c.proof.GetCodeHash(c.root, address)
+}
+
+// GetState extracts a storage slot from the witness proof for the input root hash, account address and the storage key.
+// If the witness proof contains the input storage slot for the input key, it returns its value.
+// If the proof does not contain the slot, it returns false.
+// The method may return an error if the proof is invalid.
+func (c *mptWitnessProofState) GetState(address common.Address, key common.Key) (common.Value, bool, error) {
+	return c.proof.GetState(c.root, address, key)
+}
+
+// AllStatesZero checks that all storage slots are empty for the input root hash,
+// account address and the storage key range. If the witness proof contains all empty slots
+// for the input key range, it returns true. If there is at least one non-empty slot,
+// it returns false. If the proof is not complete, it returns unknown. An incomplete proof
+// is a proof where the input address or key terminates in a node that is not a correct
+// value node, or an empty node.
+func (c *mptWitnessProofState) AllStatesZero(address common.Address, from, to common.Key) (tribool.Tribool, error) {
+	return c.proof.AllStatesZero(c.root, address, from, to)
+}
+
+// AllAddressesEmpty checks that all accounts are empty for the input root hash and the address range.
+// If the witness proof contains all empty accounts for the input address range, it returns true.
+// An empty account is an account that contains a zero balance, nonce, and code hash.
+// If there is at least one non-empty account, it returns false. If the proof is not complete,
+// it returns unknown. An incomplete proof is a proof where the input address terminates in a node
+// that is not a correct account node.
+func (c *mptWitnessProofState) AllAddressesEmpty(from, to common.Address) (tribool.Tribool, error) {
+	return c.proof.AllAddressesEmpty(c.root, from, to)
+}
```

Comment:
I dont think the documentation here is needed. It seems like a copy of the documentation of the implemented interface `witness.Proof`.

Reply:
done differently at the end